### PR TITLE
#165632384 Fix authentication class

### PR DESCRIPTION
**Pull request by engjames**
**What does this PR do?**
- Fixes the authentication error and to decode a token

**Description of the task completed**
- Decodes the token

**signup endpoint**
image

**login endpoint**
image

**get user endpoint**
image

What are the relevant pivotal tracker stories?
[#165632384](https://www.pivotaltracker.com/story/show/165632384)

### CODE REVIEW

File: authors/apps/authentication/models.py [hunk 1]
```diff
@@ -117,14 +117,14 @@ def get_short_name(self):
         return self.username
 
     @staticmethod
-    def encode_auth_token(username):
+    def encode_auth_token(email):
         """Generates auth token."""
         try:
             payload = {
                 'exp': datetime.datetime.utcnow() + datetime.timedelta(
                     days=2, seconds=20000),
```
Comment (mwinel) [inline authors/apps/authentication/models.py]: @engjames please reduce the token expiration time to atleast 15 minutes for development and testing purposes.

File: authors/apps/authentication/backends.py [hunk 2]
```diff
@@ -1,41 +1,46 @@
...
+        if prefix.lower() != 'Bearer'.lower():
+            message = "Bearer prefix missing in authorization headers."
+            raise exceptions.AuthenticationFailed(message)
+        return self.authenticate_credentials(request, token)
 
     def authenticate_credentials(self, request, token):
-        """Identify a user using the token provided"""
+        """Identify a user using the token provided."""
         try:
             payload = jwt.decode(token, settings.SECRET_KEY, 'utf-8')
```
Comment (mwinel) [inline authors/apps/authentication/backends.py:35]: @engjames we need to specifically catch a token expired error. You can use `jwt.ExpiredSignatureError` and raise a `BaseException`.

**Review by mwinel — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/backends.py (hunk 2 above)
Comment (engjames) [reply authors/apps/authentication/backends.py:35]: Thanks for that feedback am going to work on it as soon as possible

Comment (engjames) [reply authors/apps/authentication/backends.py:35]: implemented

**Review by mwinel — APPROVED** [review]
@archibishop, this should be merged.

File: authors/apps/authentication/backends.py [hunk 3]
```diff
@@ -1,41 +1,49 @@
 from django.conf import settings
+
```
Comment (sanya-kenneth) [inline authors/apps/authentication/backends.py:2]: please remove white spaces on lines 2, 4 and 11

**Review by sanya-kenneth — CHANGES_REQUESTED** [review]

File: authors/apps/authentication/backends.py (hunk 3 above)
Comment (engjames) [reply authors/apps/authentication/backends.py:2]: @sanya-kenneth Thanks for the feedback but those lines in my imports are necessary to differentiate the level of imports.

**Review by sanya-kenneth — APPROVED** [review]

**Review by 3Nakajugo — APPROVED** [review]

**Review by joelmugaya — APPROVED** [review]

**Review by archibishop — APPROVED** [review]
LGTM @engjames I am merging this.

**Review by KabohaJeanMark — COMMENTED** [review]
Great work @engjames on following through with suggested changes and fixing the bug

**Pull request merged**